src/qibo/hamiltonians/hamiltonians.py

```diff
@@ -242,11 +245,16 @@ def __mul__(self, o):
             )
         new_matrix = self.matrix * o
         r = self.__class__(self.nqubits, new_matrix, backend=self.backend)
+        o = self.backend.cast(o)
         if self._eigenvalues is not None:
             if self.backend.np.real(o) >= 0:  # TODO: check for side effects K.qnp
                 r._eigenvalues = o * self._eigenvalues
             elif not self.backend.issparse(self.matrix):
-                r._eigenvalues = o * self._eigenvalues[::-1]
+                r._eigenvalues = (
+                    o * self.backend.np.flip(self._eigenvalues, [0])
+                    if isinstance(self.backend, PyTorchBackend)
+                    else o * self._eigenvalues[::-1]
+                )
```

Comment:
```suggestion
                r._eigenvalues = o * self.backend.np.flip(self._eigenvalues, (0,))
```

Reply:
This is failing for `tensorflow`, probably because the axis are being passed as `tuple`. However, I think `torch` does not accept axis as `int`.

Reply:
Just pushed a fix for it